pkgs/by-name/cc/ccextractor/package.nix

```diff
@@ -2,85 +2,155 @@
+    inherit (finalAttrs) src;
+    sourceRoot = "${finalAttrs.src.name}/${finalAttrs.cargoRoot}";
+    patches = [ ./use-rsmpeg-0.15.patch ];
+    patchFlags = [ "-p3" ];
+    hash = "sha256-Z34QhWFs7548Cz5K176HPObv1s/VQn5VHOl7pxKOBmo=";
+  };
+
   nativeBuildInputs = [
     pkg-config
     cmake
-    makeWrapper
+    ninja
+    cargo
+    rustc
+    corrosion
+    rustPlatform.cargoSetupHook
+    rustPlatform.bindgenHook
   ];
 
   buildInputs =
-    [ zlib ]
-    ++ lib.optional (!stdenv.isLinux) libiconv
+    [
+      gpac
+      protobufc
+      libpng
+      zlib
+      utf8proc
+      freetype
+      ffmpeg_7
+      libarchive
+      curl
+      libiconv
+    ]
     ++ lib.optionals enableOcr [
       leptonica
-      tesseract4
-      ffmpeg_4
+      tesseract
     ];
 
   cmakeFlags =
     [
-      # file RPATH_CHANGE could not write new RPATH:
-      "-DCMAKE_SKIP_BUILD_RPATH=ON"
+      # TODO: This (and the corresponding patch) should probably be
+      # removed for the next stable release.
+      (lib.cmakeFeature "GIT_COMMIT_HASH" finalAttrs.src.rev)
     ]
     ++ lib.optionals enableOcr [
-      "-DWITH_OCR=on"
-      "-DWITH_HARDSUBX=on"
+      (lib.cmakeBool "WITH_OCR" true)
+      (lib.cmakeBool "WITH_HARDSUBX" true)
     ];
 
-  postInstall = lib.optionalString enableOcr ''
-    wrapProgram "$out/bin/ccextractor" \
-      --set TESSDATA_PREFIX "${tesseract4}/share/"
+  env = {
+    FFMPEG_INCLUDE_DIR = "${lib.getDev ffmpeg_7}/include";
+
+    # Upstream’s FFmpeg binding crate needs an explicit path to a shared
+    # object to do dynamic linking. The key word is *an* explicit path;
+    # they don’t support passing more than one. This linker script hack
+    # pulls in all the FFmpeg libraries they bind to.
+    #
+    # See: <https://github.com/CCExtractor/rusty_ffmpeg/pull/69>
+    FFMPEG_DLL_PATH =
+      let
+        ffmpegLibNames = [
+          "avcodec"
+          "avdevice"
+          "avfilter"
+          "avformat"
+          "avutil"
+          "swresample"
+          "swscale"
+        ];
+        ffmpegLibDir = "${lib.getLib ffmpeg_7}/lib";
+        ffmpegLibExt = stdenv.hostPlatform.extensions.library;
+        ffmpegLibPath = ffmpegLibName: "${ffmpegLibDir}/lib${ffmpegLibName}.${ffmpegLibExt}";
+        ffmpegLinkerScript = writeTextFile {
+          name = "ccextractor-ffmpeg-linker-script";
+          destination = "/lib/ffmpeg.ld";
+          text = "INPUT(${lib.concatMapStringsSep " " ffmpegLibPath ffmpegLibNames})";
+        };
+      in
+      "${ffmpegLinkerScript}/lib/ffmpeg.ld";
+  };
+
+  postPatch = lib.optionalString enableOcr ''
+    substituteInPlace src/lib_ccx/ocr.c \
+      --replace-fail 'getenv("TESSDATA_PREFIX")' '"${tesseract}/share"'
   '';
 
-  meta = with lib; {
-    homepage = "https://www.ccextractor.org";
+  meta = {
+    homepage = "https://www.ccextractor.org/";
+    changelog = "${finalAttrs.src.meta.homepage}/blob/${finalAttrs.src.rev}/docs/CHANGES.TXT";
```

Comment:
Please use the explicit link, we can way easier click on that

Reply:
Since the changelog depends on the version, there’s no way to make this a direct link. To quote the Nixpkgs manual:

> A link or a list of links to the location of Changelog for a package. A link may use expansion to refer to the correct changelog version. Example: `"https://git.savannah.gnu.org/cgit/hello.git/plain/NEWS?h=v${version}"`